### PR TITLE
Speaker API

### LINKED ISSUE #47: Speaker Screen -> API Repository
### Is your feature request related to a problem? Please describe.

Create API Repository for Speaker Screen.

### Describe the solution you'd like

1. linking the screen with API calls.

### LINKED ISSUE #56: Sponsors Screen -> API Repository
### Is your feature request related to a problem? Please describe.

Create API Repository for Sponsors Screen.

### Describe the solution you'd like

1. linking the screen with API calls.

### PR DESCRIPTION
**Pull request by TreavVasu**
### Description

Implements ApiRespository 

**Type of Change:** (Delete irrelevant options)
- Code
- Testing

**Code/Quality Assurance Only** (Delete irrelevant options)

- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

Locally Tested

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged
- [x] I have added one/more required dependencies
- [x] I have formatted the code using dart format (dartfmt)

**Code/Quality Assurance**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

**Mention the dependencies/packages/plugins that have been added (if any)**

### CODE REVIEW

File: lib/screens/speaker/speaker_repository.dart [hunk 1]
```diff
@@ -50,3 +54,33 @@ class FakeSpeakerRepository implements SpeakerRepository {
...
+    } catch (e) {
+      Log.e(tag: tag, message: "Network Error" + e.toString());
+      throw NetworkException();
+    }
+
+    if (response.statusCode == 200) {
+      Log.i(tag: tag, message: "Request Successful");
+      var speakerResponse = jsonDecode(response.body);
+      List<Speaker> speakerList;
+      (speakerResponse["data"] as List).map((e) => speakerList.add(Speaker.fromJson(e)));
```
Comment (werainkhatri) [inline lib/screens/speaker/speaker_repository.dart]: What map does is that it converts one type of datatype to another. What you are attempting to achieve is to be done using forEach.
So, you can either do this:
```suggestion
      (speakerResponse["data"] as List).forEach((e) => speakerList.add(Speaker.fromJson(e)));
```
Or this

```suggestion
      speakerList = (speakerResponse["data"] as List).map((e) => Speaker.fromJson(e)).toList();
```
Read a little more about [forEach](https://api.dart.dev/stable/1.10.1/dart-core/List/forEach.html) and [map](https://api.dart.dev/stable/1.10.1/dart-core/List/map.html) here.

File: lib/screens/speaker/speaker_repository.dart [hunk 2]
```diff
@@ -50,3 +54,33 @@ class FakeSpeakerRepository implements SpeakerRepository {
...
+class APISpeakerRepository extends SpeakerRepository {
+  final String classTag = "APIgetSpeakerRepository";
+  @override
+  Future<List<Speaker>> getAllSpeakers() async {
+    final String tag = classTag + "getAllSpeakers()";
+    http.Response response;
+    try {
+      response = await sl.get<http.Client>().get(S.getSpeakerUrl);
+    } catch (e) {
+      Log.e(tag: tag, message: "Network Error" + e.toString());
```
Comment (werainkhatri) [inline lib/screens/speaker/speaker_repository.dart]: use Log for testing. remove them when PRing your final code.

**Review by werainkhatri — CHANGES_REQUESTED** [review]

File: lib/screens/speaker/speaker_repository.dart [hunk 3]
```diff
@@ -62,17 +62,17 @@ class APISpeakerRepository extends SpeakerRepository {
     final String tag = classTag + "getAllSpeakers()";
     http.Response response;
     try {
-      response = await sl.get<http.Client>().get(S.getSpeakerUrl);
+      response = await sl.get<http.Client>().get(S.getSpeakerUrl, headers: <String, String>{
+        "Authorization": "b3c098bcf06f5b7595bffe542caea7ff41ee3587",
```
Comment (werainkhatri) [inline lib/screens/speaker/speaker_repository.dart]: I don't think there is need for Authorization token to be sent. Has anyone told you to do this?

**Review by werainkhatri — CHANGES_REQUESTED** [review]

File: lib/screens/speaker/speaker_repository.dart (hunk 3 above)
Comment (TreavVasu) [reply lib/screens/speaker/speaker_repository.dart]: Almost forgot after #84

**Review by werainkhatri — APPROVED** [review]

**Pull request merged**